scaling_performance/host_practices.adoc

```diff
@@ -0,0 +1,100 @@
+ *_/etc/origin/master/master-config.yaml_* is the local etcd instance.
+
+[[scaling-performance-capacity-host-practices-node]]
+== Recommended Practices for {product-title} Node Hosts
+
+The {product-title} node configuration file at
+*_/etc/origin/node/node-config.yaml_* contains important options such as the
+iptables synchronization period, the MTU of the SDN network, and the proxy-mode.
+
+The node configuration file allows you to pass arguments to the kubelet
+(node) process. You can view a list of possible options by running `kubelet
+--help`.
+
+[NOTE]
+====
+Not all kubelet options are supported by {product-title}.
+====
+
+One of the most commonly discussed node options
+xref:../admin_guide/manage_nodes.adoc#admin-guide-max-pods-per-node[is
+`max-pods`], which limits the number of pods that can be running on a node at
+any given time. Exceeding the `max-pods` values can result in:
+
+* Increased CPU utilization on both {product-title} and Docker.
+* Slow pod scheduling.
+* Potential out-of-memory scenarios (depends on the amount of memory in the node).
+* Exhausting the pool of IP addresses.
+* Resource overcommitting, leading to poor user application performance.
+
+[NOTE]
+====
+In Kubernetes, a pod that is holding a single
+container actually uses two containers. The second container is used to set up
+networking prior to the actual container starting. Therefore, a system running
+ten pods will actually have twenty containers running.
+====
+
+See the xref:../install_config/install/planning.adoc#sizing[Sizing Considerations] section in the installation documentation for the recommended limits for an {product-title} cluster. These limits account for the coordination between {product-title} and Docker
+for container status updates, which puts CPU pressure on the master and docker
+processes, as well as write a fair amount of log data.
+
+[[scaling-performance-capacity-host-practices-etcd]]
+== Recommended Practices for OpenShift etcd Hosts
+
+etcd is a distributed key-value store that {product-title} uses for
+configuration.
+
+After profiling etcd under {product-title}, etcd frequently performs small
+amounts of storage input and output. Using etcd with storage that handles small read/write operations quickly, such as SSD, is highly recommended. 
+
+Optimize communication between etcd and master hosts either by co-locating
+them on the same host, or providing a dedicated network.
+
+[NOTE]
+====
+etcd hosts are typically memory intensive.
+====
+
+[[scaling-performance-capacity-tuned-profile]]
+== Scaling Hosts Using the Tuned Profile
+
+Red Hat Enterprise Linux includes the `tuned` daemon, which monitors component
+activity and reacts to any changes. For example, `tuned` makes adjustments for a
+hard drive that is used heavily during startup and login, but is barely used
+when a user works with applications like OpenOffice or an email client.
+
+{product-title} includes Tuned profiles called `atomic-openshift-master` and
```

Comment:
Right. I was a little confused about that myself. @jeremyeder , any clarifications on this? Is there a difference between the Tuned profile and the tuned daemon? I got some of this info from the RHEL docs, and what i could find from google (as well as the whitepaper you gave us), so it may be confused a little.